internal/pkg/aws/secretsmanager/secretsmanager_test.go

```diff
@@ -105,3 +106,148 @@ func TestSecretsManager_CreateSecret(t *testing.T) {
 		})
 	}
 }
+
+func TestSecretsManager_DeleteSecret(t *testing.T) {
+	mockSecretName := "github-token-backend-badgoose"
+	mockError := errors.New("mockError")
+
+	tests := map[string]struct {
+		inSecretName string
+		callMock     func(m *mocks.Mockapi)
+
+		expectedError error
+	}{
+		"should wrap error returned by DeleteSecret": {
+			inSecretName: mockSecretName,
+			callMock: func(m *mocks.Mockapi) {
+				m.EXPECT().DeleteSecret(&secretsmanager.DeleteSecretInput{
+					SecretId:                   aws.String(mockSecretName),
+					ForceDeleteWithoutRecovery: aws.Bool(true),
+				}).Return(nil, mockError)
+			},
+			expectedError: fmt.Errorf("delete secret %s from secrets manager: %w", mockSecretName, mockError),
+		},
+		"should return no error if successful": {
+			inSecretName: mockSecretName,
+			callMock: func(m *mocks.Mockapi) {
+				m.EXPECT().DeleteSecret(&secretsmanager.DeleteSecretInput{
+					SecretId:                   aws.String(mockSecretName),
+					ForceDeleteWithoutRecovery: aws.Bool(true),
+				}).Return(nil, nil)
+			},
+			expectedError: nil,
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			// GIVEN
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockSecretsManager := mocks.NewMockapi(ctrl)
+			sm := SecretsManager{
+				secretsManager: mockSecretsManager,
+			}
+			tc.callMock(mockSecretsManager)
+
+			// WHEN
+			err := sm.DeleteSecret(tc.inSecretName)
+
+			// THEN
+			require.Equal(t, tc.expectedError, err)
+		})
+	}
+}
+
+func TestSecretsManager_DescribeSecret(t *testing.T) {
+	mockTime := time.Now()
+	mockSecretName := "github-token-backend-badgoose"
+	mockError := errors.New("mockError")
+	mockAPIOutput := &secretsmanager.DescribeSecretOutput{
+		CreatedDate: aws.Time(mockTime),
+		Name:        aws.String(mockSecretName),
+		Tags:        []*secretsmanager.Tag{},
+	}
+	mockOutput := &DescribeSecretOutput{
+		CreatedDate: aws.Time(mockTime),
+		Name:        aws.String(mockSecretName),
+		Tags:        []*secretsmanager.Tag{},
+	}
+	mockAwsErr := awserr.New(secretsmanager.ErrCodeResourceNotFoundException, "", nil)
+
+	tests := map[string]struct {
+		inSecretName string
+		callMock     func(m *mocks.Mockapi)
+
+		expectedResp  *DescribeSecretOutput
+		expectedError error
+	}{
+		"should wrap error returned by DescribeSecret": {
+			inSecretName: mockSecretName,
+			callMock: func(m *mocks.Mockapi) {
+				m.EXPECT().DescribeSecret(&secretsmanager.DescribeSecretInput{
+					SecretId: aws.String(mockSecretName),
+				}).Return(nil, mockError)
+			},
+			expectedError: fmt.Errorf("describe secret %s: %w", mockSecretName, mockError),
+		},
+
+		"should return no error if secret is not found": {
```

Comment:
nit: 
```suggestion
		"should return custom err if secret is not found": {
```

Reply:
🤔  but we don't error out if the secret isn't found; we only error out with other kinds of errors.